Assets/Scripts/Door.cs

```diff
@@ -1,12 +1,53 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
 
 [RequireComponent(typeof(Animator))]
 public class Door : InteractiveObject
 {
+    [Tooltip("Assigning a key here will lock the door. If the player has the key in their inventory, the can open the locked door.")]
+    [SerializeField]
+    private InventoryObject key;
+
+    [Tooltip("If this is checked, the associated key will be removed from the player's inventory when the door is unlocked")]
+    [SerializeField]
+    private bool consumesKey;
+
+    [Tooltip("The text that displays when the player looks at the door while it's locked.")]
+    [SerializeField]
+    private string lockedDisplayText = "Locked";
+
+    [Tooltip("Play this audio clip when the player interacts with a locked door without owning the key.")]
+    [SerializeField]
+    private AudioClip lockedAudioClip;
+
+    [Tooltip("Play this audio clip when the player opens the door.")]
+    [SerializeField]
+    private AudioClip openAudioClip;
+
+    // public override string DisplayText => isLocked ? lockedDisplayText : base.DisplayText;
```

Comment:
Consider removing this if the action is commented out and unnecessary. Also get rid of the accompanying blank line.

Reply:
Yes, always remove commented out code!